frame/module_configure.F

```diff
@@ -15,7 +15,8 @@ SUBROUTINE init_module_scalar_tables
   END SUBROUTINE init_module_scalar_tables
 END MODULE module_scalar_tables
 
-#if( WRF_CHEM == 1 && WRF_KPP == 1 )
+#ifdef WRF_CHEM
+#ifdef WRF_KPP
```

Comment:
Perhaps hypothetically, but without duplicating the enclosed code, how would one transform the original directive here if it involved an _or_ instead of _and_ relationship? Inability to process logical expressions in #if directives might be an issue to raise with the CMake developers.

Reply:
It would be impossible, and the current guidance (from CMake developers) is to (1) switch to Ninja or (2) rewrite the code - both of which at times are nonstarters.

This is a huge annoyance that has been noted to the CMake developers in https://gitlab.kitware.com/cmake/cmake/-/issues/17398 (which coincidentally happens to be a derivative issue spawned from https://github.com/WRF-CMake/wrf that helped me find out this limitation). 

There are a few instances in WRF already that run into this scenario, and the solution there is to do neither (1) or (2) but instead do traditional C preprocessing and generate an intermediate file. The makefile system already does this for everything; moving away from this to do native Fortran preprocessing through the respective compiler would be ideal. 

As a middle ground compromise limiting two-step intermediate file C preprocessing to only those files which absolutely need it has been an okay solution.